internal/transport/http2_client.go

```diff
@@ -291,6 +294,10 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 	if opts.MaxHeaderListSize != nil {
 		maxHeaderListSize = *opts.MaxHeaderListSize
 	}
+	maxFrameSize := defaultFrameSize
+	if opts.MaxFrameSize != nil {
```

Comment:
Don't make `maxFrameSize` or `opt.MaxFrameSize` a pointer, please, and this should validate `opts.MaxFrameSize` and clamp it in the legal range.